### PR TITLE
TF-1486 Support insert image for identity

### PR DESCRIPTION
**Pull request by dab246**
### Issue

#1486 

### Resolved

- Web:

https://github.com/linagora/tmail-flutter/assets/80730648/c67ebfc6-d026-46fa-acbe-abf6553732dd

- Mobile:

https://github.com/linagora/tmail-flutter/assets/80730648/29d1ddd9-ce1d-4ee7-9176-97c214686fed

### CODE REVIEW

File: lib/features/identity_creator/presentation/utils/identity_creator_constants.dart [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+class IdentityCreatorConstants {
+  static const int maxSizeIdentityInlineImage = 16; // Kilobyte
```
Comment (sherlockvn) [inline lib/features/identity_creator/presentation/utils/identity_creator_constants.dart:3]: IMO, should be `maxKBIdentityInlineImage` to make variable clear

File: lib/features/identity_creator/presentation/identity_creator_controller.dart [hunk 2]
```diff
@@ -456,4 +472,53 @@ class IdentityCreatorController extends BaseController {
...
+    }
+  }
+
+  bool _isExceedMaxSizeInlineImage(int fileSize) =>
+    fileSize > IdentityCreatorConstants.maxSizeIdentityInlineImage.kiloByteToBytes;
+
+  void _insertInlineImage(
+    BuildContext context,
+    PlatformFile platformFile,
+    {int? maxWith}
```
Comment (sherlockvn) [inline lib/features/identity_creator/presentation/identity_creator_controller.dart:505]: maxWith -> maxWidth

File: lib/features/identity_creator/presentation/identity_creator_controller.dart [hunk 3]
```diff
@@ -456,4 +472,53 @@ class IdentityCreatorController extends BaseController {
...
+    if (_isExceedMaxSizeInlineImage(platformFile.size)) {
+      _appToast.showToastErrorMessage(
+        context,
+        AppLocalizations.of(context).pleaseChooseAnImageSizeCorrectly(
+          IdentityCreatorConstants.maxSizeIdentityInlineImage
+        )
+      );
+    } else {
+      if (PlatformInfo.isWeb) {
+        richTextWebController.insertImageAsBase64(platformFile: platformFile);
```
Comment (sherlockvn) [inline lib/features/identity_creator/presentation/identity_creator_controller.dart:516]: you use the same method for Web and mobile platform

File: lib/features/identity_creator/presentation/utils/identity_creator_constants.dart (hunk 1 above)
Comment (dab246) [reply lib/features/identity_creator/presentation/utils/identity_creator_constants.dart:3]: > IMO, should be `maxKBIdentityInlineImage` to make variable clear

IMO, should be `maxKBSizeIdentityInlineImage` :))

File: lib/features/identity_creator/presentation/identity_creator_controller.dart (hunk 3 above)
Comment (dab246) [reply lib/features/identity_creator/presentation/identity_creator_controller.dart:516]: No, Mobile and web use 2 different editors

**Comment by chibenwa** [comment]
Can you attach here an EML of the mail with the image signature?

**Comment by dab246** [comment]
> Can you attach here an EML of the mail with the image signature?

Yeah, gift for you 

[Inline image signature.eml.zip](https://github.com/linagora/tmail-flutter/files/12035939/Inline.image.signature.eml.zip)

**Review by chibenwa — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**